docs/integrations/builtin/app-nodes/n8n-nodes-langchain.openai/text-operations.md

```diff
@@ -0,0 +1,61 @@
+---
+#https://www.notion.so/n8n/Frontmatter-432c2b8dff1f43d4b1c8d20075510fe4
+title: OpenAI Text operations 
+description: Documentation for the Text operations in OpenAI node in n8n, a workflow automation platform. Includes details of operations and configuration, and links to examples and credentials information.
+contentType: integration
+priority: critical
+---
+
+# OpenAI Text operations
+
+Use this operation to message a model or classify text for violations in OpenAI. Refer to [OpenAI](/integrations/builtin/app-nodes/n8n-nodes-base.openai/) for more information on the OpenAI node itself.
+
+## Message a Model
+
+Use this operation to send a message or prompt to an OpenAI model and receive a response.
+
+Enter these parameters:
+
+- **Credential to connect with**: Create or select an existing [OpenAI credential](/integrations/builtin/credentials/openai/).
+- **Resource**: Select **Text**.
+- **Operation**: Select **Message a Model**.
+- **Model**: Select the model you want to use. If you’re not sure which model to use then try `gpt-4o` if you need high intelligence or `gpt-4o-mini` if you need the fastest speed and lowest cost. Refer to [Models overview | OpenAI Platform](https://platform.openai.com/docs/models){:target=_blank .external-link} for more information. 
+- **Messages**: Enter a text prompt and assign a role that the model will use to generate responses. There are three roles you can choose from: 
+    - **User**: Sends a message as a user and gets a response from the model. 
+    - **Assistant**: Tells the model to adopt a specific tone or personality. 
+    - **System**: By default, the system message is `"You are a helpful assistant"`. You can define instructions in the user message, but the instructions set in the system message are more effective. You can only set one system message per conversation. Use this to set the model's behavior or context for the next user message. 
+	Refer to [Prompt engineering | OpenAI](https://platform.openai.com/docs/guides/prompt-engineering){:target=_blank .external-link} for more information on how to write a better prompt by utilizing these three roles. 
+- **Simplify Output**: Turn on to return a simplified version of the response instead of the raw data. 
+- **Output Content as JSON**: Turn on to attempt to return the response in JSON format. Compatible with `GPT-4 Turbo` and all `GPT-3.5 Turbo` models newer than `gpt-3.5-turbo-1106`.
+
+### Options
+
+- **Frequency Penalty**: Apply a penalty to reduce the model's tendency to repeat similar lines. The range is between `0.0` and `2.0`.
+- **Maximum Number of Tokens**: Set the maximum number of tokens for the response. One token is roughly four characters for standard English text. Use this to limit the length of the output. 
+- **Number of Completions**: 
```

Comment:
@ayatnkw I'm guessing this one just got lost in the mix; would you be able to add a description for this option?

Reply:
sure!